src/foam/dao/AbstractDAO.java

```diff
@@ -58,6 +58,8 @@ protected Sink decorateSink_(Sink sink, long skip, long limit, Comparator order,
       sink = new PredicatedSink(predicate, sink);
     }
 
+    ((AbstractFObject) sink).setX(x);
```

Comment:
I've updated accordingly for now. https://github.com/foam-framework/foam2/pull/558/commits/1af82c0f78fca2ed2c01401be2c53e0169a2cc47